servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcUtils.java

```diff
@@ -318,13 +295,12 @@ static <Resp> Publisher<Resp> validateResponseAndGetPayload(final StreamingHttpR
                                                                 final BufferAllocator allocator,
                                                                 final GrpcStreamingDeserializer<Resp> deserializer,
                                                                 final String httpPath) {
-        validateStatusCode(response.status()); // gRPC protocol requires 200, don't look further if this check fails.
         // In case of an empty response, gRPC-server may return only one HEADER frame with endStream=true. Our
         // HTTP1-based implementation translates them into response headers so we need to look for a grpc-status in both
         // headers and trailers. Since this is streaming response and we have the headers now, we check for the
         // grpc-status here first. If there is no grpc-status in headers, we look for it in trailers later.
         final HttpHeaders headers = response.headers();
-        validateContentType(headers, expectedContentType);
+        validateContentType(response, expectedContentType);
```

Comment:
I realized that these validations will result in a leaked undrained `StreamingHttpResponse` when they are triggered 😢 
I will send a follow-up to address that after this PR is merged